crates/biome_cli/src/reporter/gitlab.rs

```diff
@@ -0,0 +1,243 @@
+        let source_code = self.source_code(&value).unwrap_or_default();
+        let fingerprint = self
+            .ensure_fingerprint_uniqueness(self.fingerprint(&check_name, &path, &source_code), 0)
+            .to_string();
+
+        GitLabDiagnostic {
+            severity: GitLabSeverity::from(value.severity()),
+            description: PrintDescription(&value).to_string(),
+            check_name,
+            fingerprint,
+            location: Location {
+                path,
+                lines: Lines { begin: line },
+            },
+        }
+    }
+
+    /// Extracts the name of the category as the check name.
+    fn check_name(&self, value: &biome_diagnostics::error::Error) -> Option<String> {
+        let category = value.category()?;
+        Some(category.name().to_string())
+    }
+
+    /// Extracts the source code generating the diagnostic.
+    fn source_code(&self, value: &biome_diagnostics::error::Error) -> Option<String> {
+        let location = value.location();
+        let source_code = location.source_code?;
+        let span = location.span?;
+
+        Some(source_code.text[span].to_string())
+    }
+
+    /// Generates a fingerprint for a diagnostic.
+    fn fingerprint(&self, check_name: &String, path: &String, code: &String) -> u64 {
+        let mut hasher = DefaultHasher::new();
+
+        // Including the source code in our hash leads to more stable
+        // fingerprints. If you instead rely on e.g. the line number and change
+        // the first line of a file, all of its fingerprint would change.
+        code.hash(&mut hasher);
+        check_name.hash(&mut hasher);
+        path.hash(&mut hasher);
+
+        hasher.finish()
+    }
+
+    /// Enforces uniqueness of generated fingerprints in the context of a
+    /// single report.
+    fn ensure_fingerprint_uniqueness(&mut self, fingerprint: u64, salt: u64) -> u64 {
+        let mut current = fingerprint;
+        while self.fingerprints.contains(&current) {
+            let mut hasher = DefaultHasher::new();
+            current.hash(&mut hasher);
+            salt.hash(&mut hasher);
+            current = hasher.finish();
+        }
+
+        self.fingerprints.insert(current);
+        current
+    }
+
+    /// GitLab only cares about paths relative to the repository root.
+    /// This function attempts to relativize the absolute path from the
+    /// diagnostic, while falling back to the originally reported one.
+    fn path(&self, value: &biome_diagnostics::error::Error) -> Option<String> {
+        let path = match value.location().resource? {
+            biome_diagnostics::Resource::Argv => None,
+            biome_diagnostics::Resource::Memory => None,
+            biome_diagnostics::Resource::File(path) => Some(path.to_string()),
+        }?;
+
+        let Some(root) = &self.repository_root else {
+            return Some(path);
+        };
+
+        let Ok(resolved) = canonicalize(Path::new(path.as_str())) else {
```

Comment:
In the workspace, we already use this crate: https://docs.rs/path-absolutize/latest/path_absolutize/ to deal with absolute paths

You'll have to add it to the crate (`biome_cli` in this case) to the `Cargo.toml`